gemrb/plugins/SDLAudio/SDLAudio.cpp

```diff
@@ -252,22 +283,29 @@ Holder<SoundHandle> SDLAudio::Play(const char* ResRef, unsigned int channel,
 		*length = time_length;
 	}
 
-	Mix_VolumeChunk(chunk, MIX_MAX_VOLUME * GetVolume(channel) / 100);
-
-	// play
-	int chan = -1;
-	if (flags & GEM_SND_SPEECH) {
-		chan = 0;
-	}
+	Mix_VolumeChunk(chunk, MIX_MAX_VOLUME * (GetVolume(channel) * volume / 10000.0f));
 
 	std::lock_guard<std::recursive_mutex> l(OurMutex);
-	
+
 	chan = Mix_PlayChannel(chan, chunk, 0);
 	if (chan < 0) {
 		print("error playing channel");
 		return Holder<SoundHandle>();
 	}
 
+	// Positional audio
+	if (!(flags & GEM_SND_RELATIVE))
+	{
+		int x = listenerXPos - XPos;
+		int y = listenerYPos - YPos;
+		int16_t angle = atan2(y, x) * 180 / M_PI - 90;
+		if (angle < 0) {
+			angle += 360;
+		}
+		uint8_t distance = std::min(static_cast<int32_t>(sqrt(x * x + y * y) / AUDIO_DISTANCE_ROLLOFF_MOD), 255);
+		Mix_SetPosition(chan, angle, distance);
```

Comment:
I believe all our math is in radians elsewhere. It would be nice to keep it so and converting to degrees when passing into `Mix_SetPosition`

Reply:
Done I guess

Reply:
Nope. Rolled it back to angles. Mix_SetPosition has 90 degrees shift vs atan2, so we have to fix negative values after converting it to angles. It would be just cleaner too keep in degrees here